### PR TITLE
Adds RequireCodeOwnerReviews to PullRequestReviewsEnforcement

### PR DESCRIPTION
**Pull request by denniswebb**
While working on terraform-providers/terraform-provider-github#50 I realized that this was missing.

GitHub.com support page about feature https://developer.github.com/v3/repos/branches/#update-branch-protection

### CODE REVIEW

**Comment by elliott-beach** [comment]
It feels somewhat despotic to have the power to review PRs, but will proceed to do so, @gmlewis.

File: github/repos.go [hunk 1]
```diff
@@ -562,6 +562,8 @@ type PullRequestReviewsEnforcement struct {
 	DismissalRestrictions DismissalRestrictions `json:"dismissal_restrictions"`
 	// Specifies if approved reviews are dismissed automatically, when a new commit is pushed.
 	DismissStaleReviews bool `json:"dismiss_stale_reviews"`
+	// Specifies if an approved review is required in pull requests including files with a designated code owner.
```
Comment (elliott-beach) [inline github/repos.go]: To maintain the same style as the current API, this should be `RequireCodeOwnerReviews specifies...`

File: github/repos.go [hunk 2]
```diff
@@ -572,6 +574,8 @@ type PullRequestReviewsEnforcementRequest struct {
 	DismissalRestrictionsRequest *DismissalRestrictionsRequest `json:"dismissal_restrictions"`
 	// Specifies if approved reviews can be dismissed automatically, when a new commit is pushed. (Required)
 	DismissStaleReviews bool `json:"dismiss_stale_reviews"`
+	// Specifies if an approved review is required in pull requests including files with a designated code owner.
```
Comment (elliott-beach) [inline github/repos.go]: Same here.

**Comment by gmlewis** [comment]
Hi Elliott, I apologize for not discussing this with you first... are you open to reviewing PRs for this repo? You come highly recommended.  😄 
Also, just FYI, I'm about to go on vacation for 9 days and may not be very active on this repo during this time period. Others may step in to fill the gap, but if not, I'll try to catch up when I return.

**Comment by gmlewis** [comment]
Thank you!
Merging.

File: github/repos.go (hunk 1 above)
Comment (shurcooL) [reply github/repos.go]: (That is a hard requirement for top level identifiers, but for struct fields, I think both ways are acceptable. This is currently consistent with the documentation for fields above.)

**Pull request merged**